tools/generate_tables_sql_and_testNames.php

```diff
@@ -23,96 +30,210 @@
-                }
-                if($bits[0]=="select"){
-                    $bits[0]=enumizeOptions($bits[3], isset($tablename) ? $tablename : null, $bits[1]);
-                } else if($bits[0]=="selectmultiple"){
-                    $bits[0]="varchar(255)";
-                } else if($bits[0]=="textarea"){
-                    $bits[0]="text";
-                } else if($bits[0]=="text"){
-                    $bits[0]="varchar(255)";
-                } else if($bits[0]=="checkbox") {
-                    $bits[0]="varchar(255)";
-                } else if ($bits[0]=="static") {
-                    $bits[0]="varchar(255)";
-                }
-
-                $bits[2]=htmlspecialchars($bits[2]);
-                $output.="`$bits[1]` $bits[0] default NULL,\n";
+        //generate specific column definitions for specific types of HTML elements
+        default:
+            if ($bits[1] == "") {
+                continue;
+            }
+            if ($bits[0]=="select") {
+                $bits[0] = enumizeOptions(
+                    $bits[3],
+                    isset($tablename)
+                    ?? null,
+                    $bits[1]
+                );
+            } else if ($bits[0]=="selectmultiple") {
+                $bits[0] = "varchar(255)";
+            } else if ($bits[0]=="textarea") {
+                $bits[0] = "text";
+            } else if ($bits[0]=="text") {
+                $bits[0] = "varchar(255)";
+            } else if ($bits[0]=="checkbox") {
+                $bits[0] = "varchar(255)";
+            } else if ($bits[0]=="static") {
+                $bits[0] = "varchar(255)";
+            } else {
+                // Safety SQL injection prevention.
+                $bits[0] = sanitizeSqlString($bits[0]);
+            }
+            $bits[2]     = htmlspecialchars($bits[2]);
+            $output     .= "`$bits[1]` $bits[0] default NULL,\n";
+            $column_name = sanitizeSqlString($bits[1]);
+            array_push(
+                $sql_query_columns,
+                "`$column_name` $bits[0] default NULL,\n"
+            );
         }
-
     }
-    $output.="PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;\n";
-    print "Filename: $filename\n";
-    $fp=fopen($filename, "w");
-    fwrite($fp, $output);
-    fwrite($fp, "REPLACE INTO test_names (Test_name, Full_name, Sub_group) VALUES ('$tablename', '$title', 1);\n");
-    foreach($pages as $pageNo => $page) {
-        fwrite($fp, "INSERT INTO instrument_subtests (Test_name, Subtest_name, Description) VALUES ('$tablename', '" . $tablename  . "_page" . ($pageNo+1) . "', '$page');\n");
 
+    $output .= "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;\n";
+    $fp      = fopen($filename, "w");
+    fwrite($fp, $output);
+    fwrite(
+        $fp,
+        "REPLACE INTO test_names (Test_name, Full_name, Sub_group) "
+        ."VALUES ('$tablename', '$title', 1);\n"
+    );
+
+    foreach ($pages as $pageNo => $page) {
```

Comment:
A comment would be nice here explaining that this writes a record of the intended changes to a file but the file is not actually sourced directly (as was the previous behaviour)